src/bin/exrheader/main.cpp

```diff
@@ -246,21 +246,29 @@ printTimeCode (TimeCode tc)
 
 
 void
-printEnvmap (Envmap e)
+printEnvmap (const Envmap& e)
 {
-    switch (e)
+    int envmapAsInt = *( reinterpret_cast<const int*>(&e) );
```

Comment:
Is the hard core definition of default that default means "any known legal value" and is UB for unknown values? If that's the case, should we instead be declaring Envmap as an int?

i.e.

```
enum Envmap : int { ...
```

Would declaring it as such resolve the problem without all the casting?

Reply:
Changing the definition of `Envmap` does seem to resolve the problem. Is doing that an ABI change? I was trying to avoid that but maybe the simplicity of the code makes it worth doing.

Reply:
Would it be possible to do an nm on the .so with and without the  : int? If you grep the output for Envmap, we should be able to see if the mangled names with Envmap have changed or not.

Reply:
The mangled names do not change, but they move around slightly, which is a little concerning.

Will all architectures guarantee that the Envmap enum was of type `int` before this change? I'd understood that some compilers might use an 8 bit type if everything could fit in and no type was specified. If that's the case, it might cause issues in user code even if the library itself was identical

Reply:
That's inherently a problem with any sort of reinterpret_cast based solution, but I think if it was an issue it would have raised its head by now. 

If the names do not change, but move around, that's not going to cause linking problems. I'm in favor of moving to an explicit type to avoid compilers having opinions about types :)